committer/src/config.rs

```diff
@@ -68,11 +74,73 @@ pub struct App {
     pub num_blocks_to_finalize_tx: u64,
+    ///// Contains configs relating to block state posting to l1
+    pub bundle: BundleConfig,
+}
+
+/// Configuration settings for managing fuel block bundling operations.
+///
+/// This struct encapsulates various timeouts and window settings that govern
+/// how fuel blocks are accumulated, optimized, and submitted to L1.
+#[derive(Debug, Clone, Deserialize)]
+pub struct BundleConfig {
+    /// Duration to wait for additional fuel blocks before initiating the bundling process.
+    ///
+    /// This timeout is measured from the moment the last blob transaction was finalized, or, if
+    /// missing, from the application startup time.
+    ///
+    /// If no new fuel blocks are received within this period, the current set of accumulated
+    /// blocks will be bundled.
+    #[serde(deserialize_with = "human_readable_duration")]
+    pub accumulation_timeout: Duration,
+
+    /// The number of fuel blocks to accumulate before initiating the bundling process.
+    ///
+    /// If the system successfully accumulates this number of blocks before the `accumulation_timeout` is reached,
+    /// the bundling process will start immediately. Otherwise, the bundling process will be triggered when the
+    /// `accumulation_timeout` fires, regardless of the number of blocks accumulated.
+    pub blocks_to_accumulate: NonZeroUsize,
+
+    /// Maximum duration allocated for determining the optimal bundle size.
+    ///
+    /// This timeout limits the amount of time the system can spend searching for the ideal
+    /// number of fuel blocks to include in a bundle. Once this duration is reached, the
+    /// bundling process will proceed with the best configuration found within the allotted time.
+    #[serde(deserialize_with = "human_readable_duration")]
+    pub optimization_timeout: Duration,
+
+    // TODO: segfault
+    pub optimization_step: NonZeroUsize,
+
+    /// Only blocks within the `block_height_lookback` window
+    /// value will be considered for importing, bundling, fragmenting, and submitting to L1.
+    ///
+    /// This parameter defines a sliding window based on block height to determine which blocks are
+    /// eligible for processing. Specifically:
+    ///
+    /// - **Exclusion of Stale Blocks:** If a block arrives with a height less than the current
+    ///   height minus the `block_height_lookback`, it will be excluded from the bundling process.
+    ///
+    /// - **Bundling Behavior:**
+    ///   - **Unbundled Blocks:** Blocks outside the lookback window will not be bundled.
+    ///   - **Already Bundled Blocks:** If a block has already been bundled, its fragments will
+    ///     not be sent to L1.
+    ///   - **Failed Submissions:** If fragments of a bundled block were sent to L1 but failed,
+    ///     they will not be retried.
+    ///
+    /// This approach effectively "gives up" on blocks that fall outside the defined window.
+    pub block_height_lookback: u32,
```

Comment:
I'm still going over the code, but I have a question about this: Based on this bundling behavior, is there a chance that a block will be wrongfully excluded from bundling, thus not sending it to the L1?

Reply:
This depends on how many blocks are needed for fraud proving. If we only need 1w worth of blocks we'll set this config to a value bigger than what we expect in a week. 

Otherwise if it happens we need all the blocks then we'll set this to u32::MAX and will always include all blocks from genesis onwards.